packages/headless/src/api/analytics/analytics.ts

```diff
@@ -84,12 +86,8 @@ export class AnalyticsProvider implements SearchPageClientProvider {
   }
 
   public getOriginLevel2() {
-    // TODO: When tab implemented;
-    // Configurable on headless engine, optionally
-    // Need to use tabs as originLevel2, in priority if they exists/available.
-    // Otherwise, use configured originLevel2 on the engine.
-    // Ultimate fallback should be `default`;
-    return this.state.configuration.analytics.originLevel2 || 'default';
+    const {advancedSearchQueries, configuration} = this.state;
+    return advancedSearchQueries?.cq || configuration.analytics.originLevel2;
```

Comment:
It should be the opposite. 
`analytics.originLevel2;` first.

Other than that, I actually think we should not do this TODO at all.

And that we require people to set the originLevel2, through analytics actions. And no magical "resolve from that if possible"/trying to mimic JSUI.

Because setting `cq` for originLevel2 does not seem like it's solving much ? 

If you think about the end result, in UA you will get filters for your dashboard which will might be an "incomprehensible" query expression, which might be a very long string.

The goal of the tab ID in JSUI was precisely because we wanted something "human readable", to create reports on, or to get a nice URL parameter.

Reply:
Yeah, it's not ideal, but if we do not log anything, leaving it to the developers to set the value, then I suspect it is going to be a recurring situation, same as the result links, where people will not know or forget to log it.

I felt the cq was a good compromise. If you saw `@filetype=="youtubevideo"` as the originLevel2 in the analytics logs, would you say it is incomprehensible? I feel it's still readable, and has the benefit of being highly accurate, compared to if the tab id was a string like "videos" or "stuff".

The other end of the spectrum would be to introduce a new required `id` parameter when initializing a tab controller. It could be optional in v0, and switch to required in v1. Draw-backs are additional config, and the id potentially not reflecting the actual tab expression.


> It should be the opposite, analytics.originLevel2 first, I think.

That is to say, the originLevel2 should be static if configured. We can do that, but why is it desirable to configure a static originLevel2? Also, what is the use-case to configure an originLevel2 at all, when one can specific a static searchHub?

Reply:
If I'm understanding this right, with that order, `setOriginLevel2` would do nothing when a `cq` is set?

Reply:
yep, that's a good point, I'm trying to understand why developers would want to set an originLevel2 themselves though?

Reply:
I imagine for the reasons Olivier listed above, trying to set a human-readable originLevel2 when selecting a tab.

Reply:
I found [a thread in this jira](https://coveord.atlassian.net/browse/KIT-319) leaning towards letting the developer take care of setting the originLevel2.

So if we do not want to send the tab expression, then the two alternatives are:
1. Leave it to developers to set the value.
2. Expose a tab id option, and change the isActive behaviour of tabs to be based on the id rather than the cq.